tests/phpunit/TestFramework/Coverage/JUnit/JUnitTestFileDataProviderTest.php

```diff
@@ -110,4 +110,43 @@ public function test_it_works_with_feature_junit_format(): void
 
         $this->assertSame('/codeception/tests/bdd/FeatureA.feature', $testFileInfo->path);
     }
+
+    /**
+     * @dataProvider xmlProvider
+     */
+    public function test_it_does_not_trigger_count_assertion(string $xml): void
+    {
+        file_put_contents($this->tempfile, $xml);
+
+        $provider = new JUnitTestFileDataProvider($this->tempfile);
+        $provider->getTestFileInfo('ExampleTest');
+        $this->addToAssertionCount(1);
+    }
+
+    public static function xmlProvider(): iterable
```

Comment:
PHPUnit data providers don't need to be static

Reply:
Neither it needs to access `$this`. Isn't the same rule applies here as in case of static closures?

Reply:
I guess, but IIRC PHPUnit is not attempting to call it statically so I don't think there is any benefit (whereas for the static closure there is a slight perf bonus)

Reply:
Even if you call something as something non-static, it will still have a static scope.